OneCommunityGlobal/HighestGoodNetworkApp
# Jingyi leaderboard time off indicator

# Description
![image](https://github.com/user-attachments/assets/d023529a-fc8e-4c97-8040-6bf6d3f752a0)
![image](https://github.com/user-attachments/assets/7aa04e69-5d83-4312-86af-2ba6f766a4dc)

## Related PRS (if any):
This frontend PR is related to the development backend PR.

## Main changes explained:
- Implement calculation logic in Leaderboard.jsx file

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. login as **different users in one team** then schedule the timeoff currently and in the future.
5. Log in as owner/admin/manager user, navigate to the dashboard, and then to the leaderboard section to view the time off indicators.
6. Verify that names are grayed out for users who are currently on time off for the current week.
![Screenshot 2024-08-23 230207](https://github.com/user-attachments/assets/4335a1a7-1464-4cd6-9b5b-c4f7eab7083d)
7. Confirm that users with **more than the current week off** display a **"+number"** indicating additional weeks off.
8. Confirm if the user has scheduled a **future week off** then a number **without +** would be added next to name in leaderboard which would indicate the number of weeks from now for the next time off to take place for that user.
![Screenshot 2024-08-23 230200](https://github.com/user-attachments/assets/ee038c59-27dc-43c5-9d6c-83eb08587aa1)
9. Hover over the number next to a user's name to view a tooltip explaining what the number represents.
10. Finally log in as a volunteer account, make sure the volunteer role cannot see the number indicator beside the name.
![Screenshot 2024-08-23 230847](https://github.com/user-attachments/assets/d1988cc7-84bf-40e2-9712-8fc05c79a56f)

## Video of changes
https://github.com/user-attachments/assets/097bb8ad-b6f5-4ae8-9f13-612da4338324
